### PR TITLE
BUG: fix inconsistent axes ordering for axis in function `unique`

### PR DESCRIPTION
**Pull request by hgt312**
`unique` has is consistent axes order (except the chosen one) when `axis` is not `None` now.
Details in https://github.com/numpy/numpy/issues/14244.

### Change
Using `moveaxis` instead of `swapaxes` when implementation, so that the ordering
of axes except the axis in arguments will not be broken.

### CODE REVIEW

**Comment by charris** [comment]
Hmm, I assume that this changes the current behavior. Is it documented anywhere?

**Comment by hgt312** [comment]
Should I add something in docstring？

**Comment by seberg** [comment]
I think it aligns the behaviour wrt. sorting with the current documentation? So if we assume that the sort order was too strange to rely on, release notes should be sufficient, I think.

File: doc/release/upcoming_changes/14255.improvement.rst [hunk 1]
```diff
@@ -0,0 +1,4 @@
+`unique` has consistent axes order (except the chosen one) when `axis` is not None
+----------------------------------------------------------------------------------
+Using `moveaxis` instead of `swapaxes` when implementation, so that the ordering
+of axes except the axis in arguments will not be broken.
```
Comment (mattip) [inline doc/release/upcoming_changes/14255.improvement.rst]: The titile should include `numpy.unique` not just `unique

`axis`, `moveaxis`, `swapaxis` should have double-backticks so they are formatted as code, not internal links.

`"when implementation" -> "in `numpyunique`" (note single backticks around `numpy.unique`

**Pull request merged**